src/librustc_lint/builtin.rs

```diff
@@ -1605,3 +1606,39 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TrivialConstraints {
         }
     }
 }
+
+
+declare_lint! {
+    pub ELLIPSIS_INCLUSIVE_RANGE_PATTERNS,
+    Warn,
```

Comment:
cc @rust-lang/lang — do we want to warn by default when people use `...` notation (instead of the new and "shiny" `..=` notation)? Or should we tie warnings to the Edition.

Reply:
It seems like the reasons to not use `..=` are similar to the reasons to not use the edition, so an edition-tied warning (of some sort) seems reasonable.

Reply:
I guess there is a question of what our policy is about warning by default in the new edition. I think my preference would be:

1. This is an "Rust 2018 idiom" lint.
2. When you go to `edition = 2018`, all Rust 2018 Idiom lints are warn by default.
3. Perhaps in the next edition, they will become hard errors, who knows.

Reply:
That sounds great, Niko.